.github/workflows/ci-cd.yaml

```
@@ -0,0 +1,172 @@
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      vocab-importer: ${{ steps.filter.outputs.vocab-importer }}
+      backend-flask: ${{ steps.filter.outputs.backend-flask }}
+      frontend-react: ${{ steps.filter.outputs.frontend-react }}
+      opea-comps: ${{ steps.filter.outputs.opea-comps }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v2
```

actionlint (1.7.4)

the runner of "dorny/paths-filter@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)